tests/dao_tests/test_questionnaire_response_dao.py

```diff
@@ -867,6 +869,32 @@ def test_february_cope_survey(self):
         self.assertEqual(QuestionnaireStatus.SUBMITTED, participant_summary.questionnaireOnCopeFeb)
         self.assertEqual(num_completed_ppi_after_setup + 1, participant_summary.numCompletedPPIModules)
 
+    def test_covid_19_serology_results(self):
+        """ Test the covid 19 serology results survey"""
+        self.insert_codes()
+        participant_id = self.create_participant()
+        self.send_consent(participant_id)
+        questionnaire_id = self.create_questionnaire("questionnaire_covid_19_serology_results.json")
+        url = f"Participant/{participant_id}/QuestionnaireResponse"
+
+        code_answers = list()
+        code_answers.append(('covid_19_serology_results_decision', Concept(PPI_SYSTEM, 'Decision_No')))
+
+        resource = self.make_questionnaire_response_json(participant_id, questionnaire_id,
+                                                         code_answers=code_answers)
+        response = self.send_post(url, resource)
+        self.assertEqual(response["group"]["question"][0]["answer"][0]["valueCoding"]['code'], 'Decision_No')
+
+        record: ResourceData = self.session.query(ResourceData).\
+                filter(ResourceData.resourcePKAltID==participant_id).one()
+        decision_found = False
+        for mod in record.resource['modules']:
+            if mod['module'] == 'covid_19_serology_results':
+                decision_found = True
+                self.assertEqual(mod['consent_value'], 'Decision_No')
```

Comment:
Should it also verify the `mod_status` is SUBMITTED_NO_CONSENT?

Reply:
Done.